File: backend/src/services/AuditService.js
const { AuditRepository, AuditItemRepository, UserRepository } = require('../repositories');

class AuditService {
  constructor() {
    this.auditRepository = new AuditRepository();
    this.auditItemRepository = new AuditItemRepository();
    this.userRepository = new UserRepository();
  }

  async createAudit(auditData, createdBy) {
    const { title, description, status, scheduled_visit_date } = auditData;

    this.validateAuditData({ title, status, scheduled_visit_date });

    const user = await this.userRepository.getById(createdBy);
    if (!user) {
      throw new Error('Usuario creador no encontrado');
    }

    const validStatuses = ['BORRADOR', 'ENVIADO', 'APROBADO', 'RECHAZADO'];
    const finalStatus = validStatuses.includes(status) ? status : 'BORRADOR';

    const visitDate = this.validateVisitDate(scheduled_visit_date);

    const audit = await this.auditRepository.create({
      title: this.sanitizeTitle(title),
      description: description ? description.trim() : null,
      status: finalStatus,
      scheduled_visit_date: visitDate,
      created_by: createdBy
    });

    await this.createAuditItems(audit.id);

    return await this.getAuditById(audit.id);
  }

  async getAuditById(auditId) {
    if (!auditId || isNaN(auditId)) {
      throw new Error('ID de auditoría inválido');
    }

    const audit = await this.auditRepository.getWithDetails(auditId);
    if (!audit) {
      throw new Error('Auditoría no encontrada');
    }

    const items = await this.auditItemRepository.getByAuditId(auditId);
    const summary = await this.auditItemRepository.getAuditSummary(auditId);
    const scoreByCategory = await this.auditItemRepository.getAuditScoreByCategory(auditId);

    return {
      ...audit,
      items,
      summary,
      scoreByCategory,
      concept: this.calculateConcept(summary.average_score || 0)
    };
  }

  async getAllAudits(filters = {}) {
    const { status, createdBy, assignedTo, dateRange, search } = filters;

    try {
      let audits;

      if (status) {
        audits = await this.auditRepository.getByStatus(status);
      } else if (createdBy) {
        audits = await this.auditRepository.getAll({ created_by: createdBy }, 'created_at DESC');
      } else if (assignedTo) {
        audits = await this.auditRepository.getAssignedToUser(assignedTo);
      } else if (dateRange) {
        audits = await this.auditRepository.getByDateRange(dateRange.start, dateRange.end);
      } else if (search) {
        audits = await this.auditRepository.searchByTitle(search);
      } else {
        audits = await this.auditRepository.getWithCreatedBy();
      }

      return await Promise.all(
        audits.map(async (audit) => {
          const summary = await this.auditItemRepository.getAuditSummary(audit.id);
          return {
            ...audit,
            summary,
            concept: this.calculateConcept(summary.average_score || 0)
          };
        })
      );
    } catch (error) {
      throw new Error('Error al obtener auditorías: ' + error.message);
    }
  }

  async updateAudit(auditId, updateData, userId) {
    const { title, description, status, scheduled_visit_date } = updateData;

    if (!auditId || isNaN(auditId)) {
      throw new Error('ID de auditoría inválido');
    }

    const existingAudit = await this.auditRepository.getById(auditId);
    if (!existingAudit) {
      throw new Error('Auditoría no encontrada');
    }

    this.validateAuditData({ title, status, scheduled_visit_date });

    const updateFields = {};
    if (title) updateFields.title = this.sanitizeTitle(title);
    if (description !== undefined) updateFields.description = description.trim() || null;
    if (status) {
      const validStatuses = ['BORRADOR', 'ENVIADO', 'APROBADO', 'RECHAZADO'];
      if (!validStatuses.includes(status)) {
        throw new Error('Estado no válido');
      }
      updateFields.status = status;
    }
    if (scheduled_visit_date !== undefined) {
      updateFields.scheduled_visit_date = this.validateVisitDate(scheduled_visit_date);
    }

    await this.auditRepository.update(auditId, updateFields);
    return await this.getAuditById(auditId);
  }

  async deleteAudit(auditId, userId) {
    if (!auditId || isNaN(auditId)) {
      throw new Error('ID de auditoría inválido');
    }

    const audit = await this.auditRepository.getById(auditId);
    if (!audit) {
      throw new Error('Auditoría no encontrada');
    }

    if (audit.created_by !== userId) {
      throw new Error('No autorizado para eliminar esta auditoría');
    }

    await this.auditItemRepository.deleteByAuditId(auditId);
    await this.auditRepository.delete(auditId);

    return { message: 'Auditoría eliminada correctamente' };
  }

  async updateAuditScores(auditId, items, userId) {
    if (!auditId || isNaN(auditId)) {
      throw new Error('ID de auditoría inválido');
    }

    const audit = await this.auditRepository.getById(auditId);
    if (!audit) {
      throw new Error('Auditoría no encontrada');
    }

    this.validateAuditItems(items);

    await this.auditItemRepository.updateScores(auditId, items);

    const summary = await this.auditItemRepository.getAuditSummary(auditId);
    const newStatus = this.determineAuditStatus(summary.average_score || 0);
    
    await this.auditRepository.updateStatus(auditId, newStatus);

    return await this.getAuditById(auditId);
  }

  async assignAudit(auditId, userId) {
    if (!auditId || isNaN(auditId)) {
      throw new Error('ID de auditoría inválido');
    }

    const audit = await this.auditRepository.getById(auditId);
    if (!audit) {
      throw new Error('Auditoría no encontrada');
    }

    const user = await this.userRepository.getById(userId);
    if (!user) {
      throw new Error('Usuario no encontrado');
    }

    if (user.role !== 'AUDITOR') {
      throw new Error('Solo los auditores pueden ser asignados a auditorías');
    }

    const assignment = await this.auditRepository.create({
      audit_id: auditId,
      user_id: userId
    });

    return { message: 'Auditoría asignada correctamente', assignment };
  }

  async unassignAudit(auditId) {
    if (!auditId || isNaN(auditId)) {
      throw new Error('ID de auditoría inválido');
    }

    await this.auditRepository.deleteByAuditId(auditId);
    return { message: 'Asignación eliminada correctamente' };
  }

  async getAuditStats() {
    const auditStats = await this.auditRepository.getAuditStats();
    const monthlyStats = await this.auditRepository.getAuditsByMonth();

    return {
      byStatus: auditStats,
      byMonth: monthlyStats
    };
  }

  validateAuditData({ title, status, scheduled_visit_date }) {
    if (!title || title.trim().length < 3) {
      throw new Error('El título debe tener al menos 3 caracteres');
    }

    if (status) {
      const validStatuses = ['BORRADOR', 'ENVIADO', 'APROBADO', 'RECHAZADO'];
      if (!validStatuses.includes(status)) {
        throw new Error('Estado no válido');
      }
    }

    if (scheduled_visit_date && !this.isValidDate(scheduled_visit_date)) {
      throw new Error('Fecha de visita inválida');
    }
  }

  validateAuditItems(items) {
    if (!Array.isArray(items) || items.length === 0) {
      throw new Error('Se deben proporcionar items de auditoría');
    }

    for (const item of items) {
      if (!item.code || !item.score) {
        throw new Error('Cada item debe tener código y puntaje');
      }

      if (item.score < 0 || item.score > 100) {
        throw new Error('El puntaje debe estar entre 0 y 100');
      }
    }
  }

  validateVisitDate(date) {
    if (!date) return null;
    
    if (!this.isValidDate(date)) {
      throw new Error('Formato de fecha inválido. Use YYYY-MM-DD');
    }

    const visitDate = new Date(date);
    const today = new Date();
    today.setHours(0, 0, 0, 0);

    if (visitDate < today) {
      throw new Error('La fecha de visita no puede ser anterior a hoy');
    }

    return date;
  }

  isValidDate(dateString) {
    const regex = /^\d{4}-\d{2}-\d{2}$/;
    if (!regex.test(dateString)) return false;

    const date = new Date(dateString);
    return date instanceof Date && !isNaN(date);
  }

  sanitizeTitle(title) {
    return title.trim().replace(/\s+/g, ' ');
  }

  calculateConcept(averageScore) {
    if (averageScore >= 90) return 'Favorable';
    if (averageScore >= 60) return 'Favorable con requerimiento';
    return 'Desfavorable';
  }

  determineAuditStatus(averageScore) {
    if (averageScore >= 90) return 'APROBADO';
    if (averageScore >= 60) return 'ENVIADO';
    return 'RECHAZADO';
  }

  async createAuditItems(auditId) {
    const BASE_ITEMS = [
      { code: '1.1', label: 'Localización y diseño', weight: 2, category: 'INSTALACIONES FÍSICAS' },
      { code: '1.2', label: 'Condiciones de pisos y paredes', weight: 2, category: 'INSTALACIONES FÍSICAS' },
      { code: '1.3', label: 'Techos, iluminación y ventilación', weight: 2, category: 'INSTALACIONES FÍSICAS' },
      { code: '1.4', label: 'Instalaciones sanitaria', weight: 4, category: 'INSTALACIONES FÍSICAS' },
      { code: '2.1', label: 'Condiciones de equipos y utensilios', weight: 5, category: 'EQUIPOS Y UTENSILIOS' },
      { code: '2.2', label: 'Superficies de contacto con el alimento', weight: 7, category: 'EQUIPOS Y UTENSILIOS' },
      { code: '3.1', label: 'Estado de salud', weight: 7, category: 'PERSONAL MANIPULADOR' },
      { code: '3.2', label: 'Reconocimiento médico', weight: 2, category: 'PERSONAL MANIPULADOR' },
      { code: '3.3', label: 'Prácticas higiénicas', weight: 7, category: 'PERSONAL MANIPULADOR' },
      { code: '3.4', label: 'Educación y capacitación', weight: 4, category: 'PERSONAL MANIPULADOR' },
      { code: '4.1', label: 'Control de materia prima', weight: 5, category: 'REQUISITOS HIGIÉNICOS' },
      { code: '4.2', label: 'Prevención de la contaminación cruzada', weight: 9, category: 'REQUISITOS HIGIÉNICOS' },
      { code: '4.3', label: 'Manejo de temperaturas', weight: 7, category: 'REQUISITOS HIGIÉNICOS' },
      { code: '4.4', label: 'Condiciones de almacenamiento', weight: 4, category: 'REQUISITOS HIGIÉNICOS' },
      { code: '5.1', label: 'Suministro y calidad de agua potable', weight: 7, category: 'SANEAMIENTO' },
      { code: '5.2', label: 'Residuos líquidos', weight: 4, category: 'SANEAMIENTO' },
      { code: '5.3', label: 'Residuos sólidos', weight: 4, category: 'SANEAMIENTO' },
      { code: '5.4', label: 'Control integrado de plagas', weight: 9, category: 'SANEAMIENTO' },
      { code: '5.5', label: 'Limpieza y desinfección de áreas, equipos y utensilios', weight: 7, category: 'SANEAMIENTO' },
      { code: '5.6', label: 'Soportes documentales de saneamiento', weight: 2, category: 'SANEAMIENTO' },
    ];

    await this.auditItemRepository.createBulkItems(auditId, BASE_ITEMS);
  }
}

module.exports = AuditService;

File: AuditPlatform.API/Services/AuditService.cs
using AuditPlatform.API.DTOs;
using AuditPlatform.DataAccess.Context;
using AuditPlatform.Domain.Entities;
using AuditPlatform.Domain.Services;
using Microsoft.EntityFrameworkCore;

namespace AuditPlatform.API.Services;

public class AuditService
{
    private readonly ApplicationDbContext _dbContext;

    public AuditService(ApplicationDbContext dbContext)
    {
        _dbContext = dbContext;
    }

    public static bool UserCanAccessAudit(Audit audit, int userId, string? role)
    {
        if (string.Equals(role, "ADMIN", StringComparison.OrdinalIgnoreCase))
            return true;
        if (string.Equals(role, "AUDITOR", StringComparison.OrdinalIgnoreCase))
            return audit.AssignedToUserId == userId;
        return audit.CreatedBy == userId || audit.AssignedToUserId == userId;
    }

    public async Task<List<Audit>> GetAuditsForUserAsync(int userId, string? role, string? status, string? search)
    {
        var query = _dbContext.Audits
            .Include(a => a.Items)
            .Include(a => a.Evidences)
            .Include(a => a.Signatures)
            .Include(a => a.CreatedByUser)
            .Include(a => a.AssignedToUser)
            .AsQueryable();

        if (!string.Equals(role, "ADMIN", StringComparison.OrdinalIgnoreCase))
            query = query.Where(a => a.AssignedToUserId == userId || a.CreatedBy == userId);

        if (!string.IsNullOrWhiteSpace(status))
            query = query.Where(a => a.Status == status);

        if (!string.IsNullOrWhiteSpace(search))
        {
            query = query.Where(a =>
                a.Title.Contains(search) ||
                (a.InternalName ?? string.Empty).Contains(search) ||
                (a.Sede ?? string.Empty).Contains(search) ||
                (a.Description ?? string.Empty).Contains(search) ||
                (a.Comments ?? string.Empty).Contains(search));
        }

        return await query.OrderByDescending(a => a.CreatedAt).ToListAsync();
    }

    public async Task<Audit?> GetAuditForUserAsync(int id, int userId, string? role)
    {
        var audit = await _dbContext.Audits
            .Include(a => a.Items)
            .Include(a => a.Evidences)
            .Include(a => a.Signatures)
            .Include(a => a.CreatedByUser)
            .Include(a => a.AssignedToUser)
            .FirstOrDefaultAsync(a => a.Id == id);

        if (audit == null || !UserCanAccessAudit(audit, userId, role))
            return null;

        return audit;
    }

    public async Task<Audit> CreateAuditAsync(CreateAuditRequest request, int createdBy, string? creatorRole)
    {
        if (!string.Equals(creatorRole, "ADMIN", StringComparison.OrdinalIgnoreCase))
            throw new UnauthorizedAccessException("Solo los administradores pueden crear auditorías.");

        var sede = string.IsNullOrWhiteSpace(request.Sede) ? "Sin sede" : request.Sede.Trim();
        var audit = new Audit
        {
            Title = request.Title.Trim(),
            InternalName = string.IsNullOrWhiteSpace(request.InternalName) ? null : request.InternalName.Trim(),
            Sede = sede,
            Description = request.Description,
            Comments = request.Comments,
            ScheduledVisitDate = string.IsNullOrWhiteSpace(request.ScheduledVisitDate)
                ? null
                : DateTime.TryParse(request.ScheduledVisitDate, out var parsed)
                    ? parsed
                    : null,
            CreatedBy = createdBy,
            Status = "BORRADOR",
            CreatedAt = DateTime.UtcNow,
            UpdatedAt = DateTime.UtcNow
        };

        if (string.Equals(creatorRole, "ADMIN", StringComparison.OrdinalIgnoreCase)
            && request.AssignedToUserId is > 0)
        {
            var assignee = await _dbContext.Users.AsNoTracking().FirstOrDefaultAsync(u => u.Id == request.AssignedToUserId);
            if (assignee != null && assignee.Role == Role.AUDITOR)
                audit.AssignedToUserId = assignee.Id;
        }

        audit.Items = BuildActaInternaChecklist();
        _dbContext.Audits.Add(audit);
        await _dbContext.SaveChangesAsync();
        return audit;
    }

    public async Task<Audit> AssignAuditAsync(int auditId, int assignedToUserId)
    {
        var audit = await _dbContext.Audits.FirstOrDefaultAsync(a => a.Id == auditId)
                    ?? throw new KeyNotFoundException("Auditoría no encontrada.");

        var assignee = await _dbContext.Users.FirstOrDefaultAsync(u => u.Id == assignedToUserId)
                       ?? throw new InvalidOperationException("Usuario no encontrado.");

        if (assignee.Role != Role.AUDITOR)
            throw new InvalidOperationException("Solo se pueden asignar auditorías a usuarios con rol AUDITOR.");

        audit.AssignedToUserId = assignee.Id;
        audit.UpdatedAt = DateTime.UtcNow;
        await _dbContext.SaveChangesAsync();
        return audit;
    }

    public async Task<List<AuditItem>> UpdateAuditItemsAsync(int auditId, List<AuditItemUpdateRequest> items, int userId, string? role)
    {
        var audit = await _dbContext.Audits.Include(a => a.Items).FirstOrDefaultAsync(a => a.Id == auditId);
        if (audit == null)
            throw new KeyNotFoundException("Auditoría no encontrada.");

        if (!UserCanAccessAudit(audit, userId, role))
            throw new UnauthorizedAccessException("No tiene permiso para editar esta auditoría.");

        foreach (var item in items)
        {
            var existing = audit.Items.FirstOrDefault(i => i.Id == item.Id);
            if (existing == null) continue;
            existing.Score = AuditItemScoring.ClampScoreToOptions(item.Score, existing);
            existing.Observations = item.Observations;
            existing.UpdatedAt = DateTime.UtcNow;
        }

        if (AuditActaHelper.IsActaComplete(audit.Items))
            audit.Status = "CALIFICADO";

        audit.UpdatedAt = DateTime.UtcNow;
        await _dbContext.SaveChangesAsync();
        return audit.Items;
    }

    public async Task DeleteAuditAsync(int auditId, string? role)
    {
        if (!string.Equals(role, "ADMIN", StringComparison.OrdinalIgnoreCase))
            throw new UnauthorizedAccessException("Solo los administradores pueden eliminar auditorías.");

        var audit = await _dbContext.Audits
            .Include(a => a.Items)
            .Include(a => a.Evidences)
            .Include(a => a.Signatures)
            .FirstOrDefaultAsync(a => a.Id == auditId)
            ?? throw new KeyNotFoundException("Auditoría no encontrada.");

        foreach (var path in audit.Evidences.Select(e => e.FilePath)
                     .Concat(audit.Signatures.Select(s => s.ImagePath)))
            TryDeleteUploadFile(path);

        _dbContext.AuditItems.RemoveRange(audit.Items);
        _dbContext.Evidences.RemoveRange(audit.Evidences);
        _dbContext.Signatures.RemoveRange(audit.Signatures);
        _dbContext.Audits.Remove(audit);
        await _dbContext.SaveChangesAsync();
    }

    public async Task<Signature> SaveSignatureAsync(int auditId, string imageBase64, string? signerName, int userId, string? role)
    {
        var audit = await _dbContext.Audits
            .Include(a => a.Items)
            .FirstOrDefaultAsync(a => a.Id == auditId)
            ?? throw new KeyNotFoundException("Auditoría no encontrada.");

        if (!UserCanAccessAudit(audit, userId, role))
            throw new UnauthorizedAccessException("No tiene permiso para firmar esta auditoría.");

        if (!AuditActaHelper.IsActaComplete(audit.Items))
            throw new InvalidOperationException("Debe calificar todos los aspectos antes de firmar.");

        var user = await _dbContext.Users.AsNoTracking().FirstOrDefaultAsync(u => u.Id == userId);
        var displayName = string.IsNullOrWhiteSpace(signerName)
            ? user?.Name ?? "Firmante"
            : signerName.Trim();

        var bytes = DecodeBase64Image(imageBase64);
        var sigFolder = Path.Combine(Directory.GetCurrentDirectory(), "uploads", "signatures");
        Directory.CreateDirectory(sigFolder);
        var fileName = $"sig_{auditId}_{userId}_{DateTime.UtcNow:yyyyMMddHHmmss}.png";
        var relativePath = Path.Combine("uploads", "signatures", fileName);
        await File.WriteAllBytesAsync(Path.Combine(Directory.GetCurrentDirectory(), relativePath), bytes);

        var signature = new Signature
        {
            AuditId = auditId,
            SignedBy = userId,
            SignerName = displayName,
            ImagePath = relativePath.Replace('\\', '/'),
            CreatedAt = DateTime.UtcNow
        };
        _dbContext.Signatures.Add(signature);
        audit.Status = "FINALIZADO";
        audit.UpdatedAt = DateTime.UtcNow;
        await _dbContext.SaveChangesAsync();
        return signature;
    }

    public async Task<List<Signature>> GetSignaturesAsync(int auditId, int userId, string? role)
    {
        var audit = await _dbContext.Audits.AsNoTracking().FirstOrDefaultAsync(a => a.Id == auditId);
        if (audit == null || !UserCanAccessAudit(audit, userId, role))
            throw new KeyNotFoundException("Auditoría no encontrada.");

        return await _dbContext.Signatures
            .Where(s => s.AuditId == auditId)
            .OrderBy(s => s.CreatedAt)
            .ToListAsync();
    }

    public async Task<Audit?> GetAuditForPdfAsync(int id, int userId, string? role)
    {
        var audit = await _dbContext.Audits
            .Include(a => a.Items.OrderBy(i => i.Code))
            .Include(a => a.Signatures)
            .Include(a => a.CreatedByUser)
            .Include(a => a.AssignedToUser)
            .FirstOrDefaultAsync(a => a.Id == id);

        if (audit == null || !UserCanAccessAudit(audit, userId, role))
            return null;

        if (!AuditActaHelper.IsActaComplete(audit.Items))
            throw new InvalidOperationException("El acta debe estar calificada por completo para generar el PDF.");

        return audit;
    }

    private static byte[] DecodeBase64Image(string input)
    {
        var data = input.Trim();
        var comma = data.IndexOf(',');
        if (comma >= 0)
            data = data[(comma + 1)..];
        return Convert.FromBase64String(data);
    }

    private static void TryDeleteUploadFile(string relativePath)
    {
        if (string.IsNullOrWhiteSpace(relativePath)) return;
        var full = Path.Combine(Directory.GetCurrentDirectory(), relativePath.Replace('/', Path.DirectorySeparatorChar));
        if (File.Exists(full))
            File.Delete(full);
    }

    public async Task<List<Evidence>> GetEvidenceByAuditAsync(int auditId, int userId, string? role)
    {
        var audit = await _dbContext.Audits.AsNoTracking().FirstOrDefaultAsync(a => a.Id == auditId);
        if (audit == null || !UserCanAccessAudit(audit, userId, role))
            throw new KeyNotFoundException("Auditoría no encontrada.");

        return await _dbContext.Evidences.Where(e => e.AuditId == auditId).ToListAsync();
    }

    public async Task<Evidence> AddEvidenceAsync(int auditId, string filePath, int userId, string? role)
    {
        var audit = await _dbContext.Audits.FirstOrDefaultAsync(a => a.Id == auditId);
        if (audit == null || !UserCanAccessAudit(audit, userId, role))
            throw new KeyNotFoundException("Auditoría no encontrada.");

        var evidence = new Evidence
        {
            AuditId = auditId,
            FilePath = filePath,
            CreatedAt = DateTime.UtcNow
        };

        _dbContext.Evidences.Add(evidence);
        await _dbContext.SaveChangesAsync();
        return evidence;
    }

    public async Task<AuditStats> GetAuditStatsAsync()
    {
        var allAudits = await _dbContext.Audits.ToListAsync();
        var total = allAudits.Count;
        var approved = allAudits.Count(a => a.Status == "APROBADO");
        var rejected = allAudits.Count(a => a.Status == "RECHAZADO");
        var drafts = allAudits.Count(a => a.Status == "BORRADOR");
        var completionRate = total == 0 ? 0 : Math.Round((double)(approved + rejected) / total * 100, 2);

        return new AuditStats(total, approved, rejected, drafts, completionRate);
    }

    public async Task<DashboardMetrics> GetDashboardMetricsAsync()
    {
        var audits = await _dbContext.Audits.ToListAsync();
        var byStatus = audits.GroupBy(a => a.Status).ToDictionary(g => g.Key, g => g.Count());

        return new DashboardMetrics(audits.Count, byStatus);
    }

    /// <summary>
    /// Plantilla alineada al formato "Acta de auditoría interna" (CUZCO / Carrascal Zamarra): ítems con calificación A / AR / I.
    /// </summary>
    private static List<AuditItem> BuildActaInternaChecklist()
    {
        static AuditItem Row(string code, string label, string category, double a, double ar, double i)
            => new()
            {
                Code = code,
                Label = label,
                Category = category,
                PointsA = a,
                PointsAR = ar,
                PointsI = i,
                Weight = a,
                Score = -1
            };

        return
        [
            Row("1.1", "Localización y diseño", "1. INSTALACIONES FÍSICAS", 7, 3.5, 0),
            Row("1.2", "Condiciones de pisos y paredes", "1. INSTALACIONES FÍSICAS", 7, 3.5, 0),
            Row("1.3", "Techos, iluminación y ventilación", "1. INSTALACIONES FÍSICAS", 7, 3.5, 0),
            Row("1.4", "Instalaciones sanitarias", "1. INSTALACIONES FÍSICAS", 4, 2, 0),
            Row("2.1", "Condiciones de equipos y utensilios", "2. EQUIPOS Y UTENSILIOS", 6, 3, 0),
            Row("2.2", "Superficies de contacto con el alimento", "2. EQUIPOS Y UTENSILIOS", 6, 3, 0),
            Row("3.1", "Estado de salud", "3. PERSONAL MANIPULADOR", 4, 2, 0),
            Row("3.2", "Reconocimiento médico", "3. PERSONAL MANIPULADOR", 4, 2, 0),
            Row("3.3", "Prácticas higiénicas", "3. PERSONAL MANIPULADOR", 7, 3.5, 0),
            Row("3.4", "Educación y capacitación", "3. PERSONAL MANIPULADOR", 4, 2, 0),
            Row("4.1", "Control de materia prima", "4. REQUISITOS HIGIÉNICOS", 5, 2.5, 0),
            Row("4.2", "Prevención de la contaminación cruzada", "4. REQUISITOS HIGIÉNICOS", 5, 2.5, 0),
            Row("4.3", "Manejo de temperaturas", "4. REQUISITOS HIGIÉNICOS", 7, 3.5, 0),
            Row("4.4", "Condiciones de almacenamiento", "4. REQUISITOS HIGIÉNICOS", 4, 2, 0),
            Row("5.1", "Suministro y calidad de agua potable", "5. SANEAMIENTO", 7, 3.5, 0),
            Row("5.2", "Residuos líquidos", "5. SANEAMIENTO", 4, 2, 0),
            Row("5.3", "Residuos sólidos", "5. SANEAMIENTO", 4, 2, 0),
            Row("5.4", "Control integrado de plagas", "5. SANEAMIENTO", 9, 4.5, 0),
            Row("5.5", "Limpieza y desinfección de áreas, equipos y utensilios", "5. SANEAMIENTO", 7, 3.5, 0),
            Row("5.6", "Soportes documentales de saneamiento", "5. SANEAMIENTO", 2, 1, 0),
        ];
    }
}
